File: app/src/main/java/com/sudoajay/pdf_viewer/SharedPreference/SdCardPathSharedPreference.java
package com.sudoajay.pdf_viewer.SharedPreference;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;


import com.sudoajay.pdf_viewer.R;

import static android.content.Context.MODE_PRIVATE;

public class SdCardPathSharedPreference {

    // global varibale
    private SharedPreferences.Editor editor;
    private Context context;
    private SharedPreferences pref;


    // constructor
    @SuppressLint("CommitPrefEdits")
    public SdCardPathSharedPreference(Context context) {
        pref = context.getSharedPreferences(context.getString(R.string.MY_PREFS_NAME), MODE_PRIVATE);
        editor = pref.edit();
        this.context = context;

        // default value pass
        // grab the data from shared preference
        editor.putString(context.getString(R.string.sdCardPath), "");
        editor.putString(context.getString(R.string.stringUri), "");

    }

    public String getSdCardPath() {
        return pref.getString(context.getString(R.string.sdCardPath), "");
    }

    public void setSdCardPath(String sdCardPath) {

        // send thd data to shared preferences
        editor.putString(context.getString(R.string.sdCardPath), sdCardPath);
        editor.apply();
    }

    public String getStringURI() {
        return pref.getString(context.getString(R.string.stringUri), "");
    }

    public void setStringURI(String stringURI) {

        editor.putString(context.getString(R.string.stringUri), stringURI);
        editor.apply();
    }
}

File: app/src/main/java/com/sudoajay/pdf_viewer/sharedPreference/SdCardPathSharedPreference.kt
package com.sudoajay.pdf_viewer.sharedPreference

import android.content.Context
import android.content.SharedPreferences
import android.content.SharedPreferences.Editor
import com.sudoajay.pdf_viewer.R

class SdCardPathSharedPreference constructor(context: Context) {
    // global variable
    private val editor: Editor
    private val context: Context
    private val pref: SharedPreferences = context.getSharedPreferences(context.getString(R.string.MY_PREFS_NAME), Context.MODE_PRIVATE)
    // send thd data to shared preferences
    var sdCardPath: String
        get() = pref.getString(context.getString(R.string.sdCardPath), "").toString()
        set(sdCardPath) { // send thd data to shared preferences
            editor.putString(context.getString(R.string.sdCardPath), sdCardPath)
            editor.apply()
        }

    var stringURI: String
        get() = pref.getString(context.getString(R.string.sdCardStringUri), "").toString()
        set(stringURI) {
            editor.putString(context.getString(R.string.sdCardStringUri), stringURI)
            editor.apply()
        }

    // constructor
    init {
        editor = pref.edit()
        editor.apply()
        this.context = context

    }
}